src/js/ctrl/export.ts

```diff
@@ -216,12 +227,16 @@ exporter.mark = function(state: State, internal: boolean, id: number) {
     spec.from = {data: facet.name};
   } else if (spec.from) {
     let fromId;
-    if ((fromId = spec.from.data)) {
+    if ((fromId = spec.from.name)) {
```

Comment:
i'm not a fan of assignment inside of a boolean expression for code readability purposes. i know you're just following the pattern that was already here, but would you mind re-writing these if cases to be more explicit?
e.g.
```
if (spec.from.name) {
 ...
}
else if (spec.from.data) {
  ... 
}
else if (spec.from.mark) {
 ...
}
else if (spec.from.facet.data) {
 ...
}